rustiflow/src/main.rs

```diff
@@ -1004,54 +954,24 @@ fn process_packet_ipv4<T>(
         window_size: data.window_size,
     };
     let flow_id = if fwd {
-        create_flow_id(
-            &source,
-            data.port_source,
-            &destination,
-            data.port_destination,
-            data.protocol,
-        )
+        create_flow_id(&source, data.port_source, &destination, data.port_destination, data.protocol)
     } else {
-        create_flow_id(
-            &destination,
-            data.port_destination,
-            &source,
-            data.port_source,
-            data.protocol,
-        )
+        create_flow_id(&destination, data.port_destination, &source, data.port_source, data.protocol)
     };
 
-    let flow_id_clone = flow_id.clone();
-    let flow_id_remove = flow_id.clone();
-    let mut entry = flow_map.entry(flow_id).or_insert_with(|| {
+    let mut entry = flow_map.entry(flow_id.clone()).or_insert_with(|| {
         if fwd {
-            T::new(
-                flow_id_clone,
-                source,
-                data.port_source,
-                destination,
-                data.port_destination,
-                data.protocol,
-                ts_date,
-            )
+            T::new(flow_id.clone(), source, data.port_source, destination, data.port_destination, data.protocol, ts_date)
         } else {
-            T::new(
-                flow_id_clone,
-                destination,
-                data.port_destination,
-                source,
-                data.port_source,
-                data.protocol,
-                ts_date,
-            )
+            T::new(flow_id.clone(), destination, data.port_destination, source, data.port_source, data.protocol, ts_date)
         }
     });
 
     let end = entry.update_flow(&features, &ts, ts_date, fwd);
-    if end.is_some() {
-        export(&end.unwrap());
+    if let Some(flow) = end {
+        export(&flow);
```

Comment:
We should add the same if statement for non contaminant export also.
```rs
            if *NO_CONTAMINANT_FEATURES.lock().unwrap().deref() {
                export(&flow.dump_without_contamination());
            } else {
                export(&flow.dump());
            }
```

Reply:
Good catch, fixed!